testing/run_ap_with_constraint.py

```diff
@@ -42,40 +44,42 @@ def run_test_main(args):
     # check if test suite and test directory exist. 
     test_path = os.path.join(args.test_cases_path, args.test_suite_name)
     if(not os.path.isdir(test_path)):
-        print(test_path, " does not exists!")
+        print(f"Error: {test_path} does not exists!")
+        return False
+
+    print(f"Running AP flow using intermediate files located at: {test_path}")
+
+    ap_run_base_dir = os.path.join(args.runs_path, args.test_suite_name)
+    os.makedirs(ap_run_base_dir, exist_ok=True)
+    runs = [d for d in os.listdir(ap_run_base_dir) if os.path.isdir(os.path.join(ap_run_base_dir, d)) and d.startswith("run")]
+    last_run_num = 0
+    if len(runs) is not 0:
+        last_run = sorted(runs)[-1]
+        last_run_num = extract_run_number(last_run)
+        if last_run_num is None:
+            last_run_num = 0
+    run_num = last_run_num + 1
+    run_name = "run{:03d}".format(run_num)
+    run_dir = os.path.join(ap_run_base_dir, run_name)
+    os.makedirs(run_dir, exist_ok=True)
+    print(f"Saving run information to: {run_dir}")
```

Comment:
This change would affect the collect_hpwl script as well, since that expects the run### be the inner directory.

Reply:
Thanks for the catch. Fixed.